### DISCUSSION TITLE
Programatic access

### DISCUSSION THREAD

**Discussion by frasou** [christgau/wsdd#87] [Q&A] [answered]
Hello, 
thank you for this tool, it really helped me.

However, I am not able to find a unix socket of running wsdd. So I can not ask wsdd about discovered peers or initiate probe. There are only several broken links in /proc/PID_of_wsdd/fd/. Could you point me into the right direction, please?

**Comment by christgau** [comment]
If you do not specify a path for a unix socket then a locally bound tcp socket is opened. There is also a warning for this
```
$ ./wsdd.py -D
2021-02-23 21:36:23,167:wsdd WARNING(pid 6278): no interface given, using all interfaces
2021-02-23 21:36:23,226:wsdd WARNING(pid 6278): Discovery enabled but no listen option provided. Falling back to port 5359
```
You can specify a unix domain socket by providing a path with the `-l/--listen` option. If the path is numeric it will be interpreted as TCP port number.

**Comment by frasou** [comment, marked as answer]
Thanks,

i am sorry, i was not specific enough. I have read you manual, it is clearly described. What I don't know how to pass a command list/clear/prune to the wsdd.

`wsdd -v -D -l /tmp/test-socket list`

`usage: wsdd [-h] [-i INTERFACE] [-H HOPLIMIT] [-U UUID] [-v] [-d DOMAIN] [-n HOSTNAME] [-w WORKGROUP] [-t] [-4] [-6] [-s] [-p] [-c CHROOT] [-u USER] [-D] [-l LISTEN] [-o] [-V]
wsdd: error: unrecognized arguments: list`

**Reply by christgau** [reply]
You have to connect to the socket that is opened by wsdd following what you have specified on the command line. In your case you have a unix domain socket under `/tmp/test-socket`. To connect to that socket you have to use some software that is able to do so, e.g. netcat (the OpenBSD variant) or socat, or you could write your own client.

For socat, you can do the following: `echo list | socat - UNIX-CONNECT:/tmp/test-socket`

Note that you have to reconnect to the socket after your request since the connection is closed by wsdd after each request. This a known limitation of the current code. Please also note that you have to delete the unix domain socket after wsdd has exited, also a limitation of the current code.

**Reply by frasou** [reply]
Hello, thank you very much, I didn't know how to work with a unix socket. And it works flawlessly.
